backend/src/util.rs

```diff
@@ -0,0 +1,32 @@
+use load::Data;
+use chrono::NaiveDateTime;
+
+/// Returns where the passed date is or should go in the sorted data slice.
+fn get_insert_location(data: &[Data], date: &NaiveDateTime) -> ::std::result::Result<usize, usize> {
+    data.binary_search_by(|probe| probe.date.cmp(date))
+}
+
+/// Return the start index for an iterator from the passed date to the index
+/// returned by the companion function, `end_idx`.
+pub fn start_idx(data: &[Data], date: &NaiveDateTime) -> usize {
+    match get_insert_location(data, date) {
+        Ok(idx) => idx,
+        Err(idx) => if idx != 0 {
+            idx - 1
+        } else {
+            0
+        }
+    }
+}
+
+/// Returns the end index for an iterator from the `start_idx()` to this date.
+pub fn end_idx(data: &[Data], date: &NaiveDateTime) -> usize {
+    match get_insert_location(data, date) {
+        Ok(idx) => idx,
+        Err(idx) => if idx < data.len() {
+            idx
+        } else {
+            data.len() - 1
+        }
+    }
+}
```

Comment:
Both `start_idx` and `end_idx` basically just clamp the value into the bounds of 0..data.len(); is there something in the stdlib better than what we do now?